### PR TITLE
Enable experimental support for lsp within the shell-scripts layer

### PR DESCRIPTION
**Pull request by DanielRichtmann**
Note that this is only valid for bash scripts, since it uses the
`bash-language-server` defined in `lsp-clients.el`

Adds LSP support for `bash` shell scripts.

Inspired by the python implementation.

Completion and jump to definition are working in first tests.

Still TODO:

Further testing

Since this is my first work on `spacemacs` and also coding in `elisp` I'm grateful for all comments.

### CODE REVIEW

File: layers/+lang/shell-scripts/funcs.el [hunk 1]
```diff
@@ -18,3 +18,35 @@
...
+      (lsp)
+    (message "`lsp' layer is not installed, please add `lsp' layer to your dotfile.")))
+
+(defun spacemacs//shell-scripts-setup-company ()
+  "Conditionally setup company based on backend."
+  (message "%s setting up company" shell-scripts-backend)
+  (pcase shell-scripts-backend
+    (`lsp (spacemacs//shell-scripts-setup-lsp-company))))
+
+(defun spacemacs//shell-scripts-setup-lsp-company ()
```
[the remark this answers is not in the document]
Comment (DanielRichtmann) [reply layers/+lang/shell-scripts/funcs.el]: This is verbatim the same comment as in the php PR:

I now removed the company-related stuff, how does it look to you now?

Please also note, that I'm explicitly keeping the commits here for documentation purposes. I'm going to squash these once this PR is ready to be cherry-picked.

**Comment by robbyoconnor** [comment]
You need to update the `CHANGELOG.develop` file :)

**Comment by smile13241324** [comment]
Hi guys,

Has someone time to test this and post some results for bash. This would speed things up considerably.

@robbyoconnor

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs :purple_heart:, I have squashed and cherry picked your commits into develop.

**Pull request closed without merging**